src/NHibernate.Test/CacheTest/FilterKeyFixture.cs

```diff
@@ -8,105 +9,129 @@ namespace NHibernate.Test.CacheTest
 	[TestFixture]
 	public class FilterKeyFixture: TestCase
 	{
-		protected override string MappingsAssembly
-		{
-			get{return "NHibernate.Test";}
-		}
+		protected override string MappingsAssembly => "NHibernate.Test";
 
-		protected override IList Mappings
-		{
-			get { return new[] { "CacheTest.EntityWithFilters.hbm.xml" }; }
-		}
+		protected override IList Mappings => new[] { "CacheTest.EntityWithFilters.hbm.xml" };
 
 		[Test]
 		public void ToStringIncludeAll()
 		{
 			string filterName = "DescriptionLike";
 			var f = new FilterImpl(Sfi.GetFilterDefinition(filterName));
 			f.SetParameter("pLike", "so%");
-			var fk = new FilterKey(filterName, f.Parameters, f.FilterDefinition.ParameterTypes);
-			Assert.That(fk.ToString(), Is.EqualTo("FilterKey[DescriptionLike{'pLike'='so%'}]"));
+			var fk = new FilterKey(f);
+			Assert.That(fk.ToString(), Is.EqualTo("FilterKey[DescriptionLike{'pLike'='so%'}]"), "Like");
 
 			filterName = "DescriptionEqualAndValueGT";
 			f = new FilterImpl(Sfi.GetFilterDefinition(filterName));
 			f.SetParameter("pDesc", "something").SetParameter("pValue", 10);
-			fk = new FilterKey(filterName, f.Parameters, f.FilterDefinition.ParameterTypes);
-			Assert.That(fk.ToString(), Is.EqualTo("FilterKey[DescriptionEqualAndValueGT{'pDesc'='something', 'pValue'='10'}]"));
+			fk = new FilterKey(f);
+			Assert.That(fk.ToString(), Is.EqualTo("FilterKey[DescriptionEqualAndValueGT{'pDesc'='something', 'pValue'='10'}]"), "Value");
 		}
 
 		[Test]
 		public void Equality()
 		{
-			// Equality is aware only by parameters names not values
-			FilterKey fk, fk1;
-			FilterDescLikeToCompare(out fk, out fk1);
-			Assert.That(fk, Is.EqualTo(fk1));
+			FilterDescLikeToCompare(out var fk, out var fk1, true);
```

Comment:
They can be equal only with same value. The `NotEquality` test is completed with an assert showcasing that a different value causes inequality.